### PR TITLE
(GH-145) Add validation mode flag to puppetized DSC Resources

### PR DESCRIPTION
**Pull request by michaeltlombardi**
This PR adds a `validation_mode` flag to the puppetized resources which changes their behavior from Puppet's perspective; by default resources will use the `property` validation mode, which relies on Puppet's own property insync validation to tell whether or not a resource is in the desired state. If the manifest writer instead specifies the validation mode to be `resource`, it will call `Invoke-DscResource` with the `Test` method and trust that result, reporting in/out of state for **all** properties of the resource. This results in less granular change reporting but reduces the incidence of flapping, failing to recognize changes correctly, etc.

This PR adds documentation for using the validation mode flag and tests to validate the changes in behavior.

This change is blocked on the resolving of puppetlabs/puppet-resource_api#225 and its subsequent release into Puppet, as well as a PDK version which bundles those versions into itself.

It is also blocked on the merging of puppetlabs/ruby-pwsh#124 which implements the custom insync logic in the base provider.

TODO:

- [x] Write a changelog entry
- [x] Ensure tests function as expected

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
Blocked on the next releases of Puppet (`6.23.0` and `7.?.?` and the PDK which incorporates them).

**Comment by michaeltlombardi** [comment]
The implementation here has been validated against the implementation of custom insync in the DSC Base provider (as seen in the passing acceptance tests against a git reference). Once that implementation is merged and a release is cut, the commit modifying the git reference for acceptance should be dropped, at which point this PR can be merged and #145 will be resolved.

**Review by david22swan — APPROVED** [review]

**Pull request merged**
